Massive documentation fixes + copy editing (#15747)
nim-lang/Nim: doc/tut1.rst

  nim c -d:release greetings.nim

By default, the Nim compiler generates a large number of runtime checks
aiming for your debugging pleasure. With ``-d:release`` some checks are
`turned off and optimizations are turned on

> <nimc.html#compiler-usage-compile-time-symbols>`_.

Reviewer: ```suggestion
<nimc.html#compiler-usage-compileminustime-symbols>`_.
```
There's also a similar reference in lib/system.nim, line ~99.

Every variable, procedure, etc. needs to be declared before it can be used.
(The reason for this is that it is non-trivial to avoid this need in a

> language that supports meta-programming as extensively as Nim does.)

Reviewer 2: ```suggestion
language that supports metaprogramming as extensively as Nim does.)
```

I think that in most cases Nim docs/tutorials/blog posts refer to metaprogramming without any hyphens

Author: I had my doubts about this one as well, tbh...

  ``yield`` statement).
* Iterators have no implicit ``result`` variable.
* Iterators do not support recursion.
* Iterators cannot be forward declared, because the compiler must be able to inline an iterator. (This restriction will be gone in a
  future version of the compiler.)

However, you can also use a ``closure`` iterator to get a different set of

> restrictions. See `first-class iterators <manual.html#iterators-and-the-for-statement-first-class-iterators>`_

Reviewer: ```suggestion
restrictions. See `first-class iterators <manual.html#iterators-and-the-for-statement-firstminusclass-iterators>`_
```
See my other comment for a more detailed explanation.
